app/lib/app/SecvisogramPage/View/WizzardEditorTab/WizardPanel/editors/GenericEditor/ArrayEditor.js

```diff
@@ -69,44 +123,49 @@ export default function ArrayEditor({ property, instancePath }) {
                   type="button"
                   className="w-9 h-9 flex-none hover:bg-gray-300"
                   onClick={() => {
-                    sideBarData.setSideBarIsOpen(true)
-                    sideBarData.setSideBarSelectedPath(instancePath.concat(i.toString()))
+                    setSideBarIsOpen(true)
+                    setSideBarSelectedPath(instancePath.concat(i.toString()))
                   }}
                 >
                   <FontAwesomeIcon icon={faInfoCircle} size="xs" />
                 </button>
-              </li>
-            )
-          })}
-        </ul>
-        <button
-          type="button"
-          className="h-9 px-2 text-center hover:bg-gray-300 border-b border-gray-300 border-solid w-full"
-          onClick={() => {
-            const value =
-              childProperty.type === 'OBJECT'
-                ? {}
-                : childProperty.type === 'ARRAY'
-                ? []
-                : ['STRING', 'DATETIME', 'URI'].includes(childProperty.type)
-                ? ''
-                : null
-            if (value !== null) {
-              updateDoc(instancePath, sanitizedValue.concat([value]))
-            }
-          }}
-        >
-          Add item
-        </button>
-      </div>
-      {typeof selectedIndex === 'number' ? (
-        <GenericEditor
-          key={instancePath.concat([String(selectedIndex)]).join('.')}
-          parentProperty={property}
-          property={childProperty}
-          instancePath={instancePath.concat([String(selectedIndex)])}
-        />
-      ) : null}
-    </>
+              </div>
+              {recursionProperty ? (
+                <Menu
+                  {...props}
+                  instancePath={instancePath.concat([
+                    String(i),
+                    recursionProperty.key,
+                  ])}
+                  level={level + 1}
+                />
+              ) : null}
+            </li>
+          )
+        })}
+      </ul>
+      <button
+        type="button"
+        className={`h-9 pr-2 text-left hover:bg-gray-300 w-full whitespace-nowrap ${
+          level > 1 ? 'pl-4' : 'pl-2'
+        }`}
+        data-testid={`menu_entry-/${instancePath.join('/')}-add_item_button`}
+        onClick={() => {
+          const value =
+            childProperty.type === 'OBJECT'
+              ? {}
+              : childProperty.type === 'ARRAY'
+              ? []
+              : ['STRING', 'DATETIME', 'URI'].includes(childProperty.type)
+              ? ''
+              : null
+          if (value !== null) {
+            updateDoc(instancePath, sanitizedValue.concat([value]))
+          }
+        }}
+      >
+        Add item
```

Comment:
Eventually, we might want to have an alternative wording here in case of recursion. Off the top of my head I could imagine sth. like

```javascript
{recursionProperty ? "Add child" : "Add item"}
```

Reply:
I agree. We decided that this will be done in a new Branch dedicated to styling.